### PR TITLE
Feature/data handler2

### PR DESCRIPTION
**Pull request by layertwo**
Here's the PR for the updates to DataHandler.

I think most of this may already be in the development branch, but figured it's worth it to open this PR for discussion.

Comments on code will be inline.

### CODE REVIEW

File: app/data_handler.py [hunk 1]
```diff
@@ -21,12 +21,17 @@ def addHostToDB(self, hostname, ipv4_addr, type, ios_type, local_creds):
...
-                                   type=type, ios_type=ios_type,
+                                   type=type.capitalize(),
+                                   ios_type=ios_type,
                                    local_creds=local_creds)
             app.db.session.add(host)
             # This enables pulling ID for newly inserted host
             app.db.session.flush()
             app.db.session.commit()
+        except (IntegrityError, InvalidRequestError) as e:
+            return False, 0, e
```
Comment (layertwo) [inline app/data_handler.py]: should there be an app.db.session.rollback() here?

File: app/data_handler.py [hunk 2]
```diff
@@ -78,21 +94,20 @@ def importHostsToDB(self, csvImport):
...
-                host = app.models.Host(hostname=row[0], ipv4_addr=row[1],
+                # TODO could probably use self.addHostToDB
+                host = app.models.Host(hostname=row[0].strip(),
+                                       ipv4_addr=row[1],
                                        type=row[2].capitalize(),
                                        ios_type=ios_type,
                                        local_creds=local_creds)
                 app.db.session.add(host)
                 app.db.session.flush()
+                app.db.session.commit()
```
Comment (layertwo) [inline app/data_handler.py]: as a consideration, we should probably limit committing to the database for each host, and should just batch them all up and do it once.

Comment (matt852) [reply app/data_handler.py]: Good suggestion.  Added to this PR.

File: app/data_handler.py (hunk 1 above)
Comment (matt852) [reply app/data_handler.py]: Yes.  Added

**Review by matt852 — APPROVED** [review]

**Pull request merged**